### PR TITLE
Refactor relationships

### PR DESCRIPTION
**Pull request by djaiss**
This is a massive change of the codebase, including some breaking changes in the API.

This is the first step to address #19 (set relationship to a person) and https://github.com/monicahq/monica/issues/117.

The idea is 
* to merge kids and partners into a single table
* to create a new table `relationship_types` which would allow any type of relationships between two persons.

The scope of this PR **is not** to allow custom relationships now. It is to have the equivalent of what we have now, with the new tables structure, make sure it works. It has to be transparent to the user.

The new system manages masculine/feminine of each relationship type (and localized too), like `Father/mother`, `daughter/son`, etc...

Each new relationship type belongs to a relationship type group. For instance, `daughter/son` type belongs to the `Family` relationship type group.

Here are the relationship types that we are going to introduce:
* Partner
* Spouse
* Date
* Lover
* In love with
* Ex
* Parent
* Child
* Sibling
* Grand parents
* Grand children
* Uncle
* Nephew
* Cousine
* God father
* God son
* Friend
* Best friend
* Colleague
* Boss
* Mentor

#### Remaining tasks

- [X] Create new relationship table
- [X] Migrate current significant others to the new system
- [X] Update the add/edit/delete relationship views
- [x] Migrate current kids/offsprings/parents into the new Relationships table in a migration
- [x] Put back reminders and make them work with the new system
- [x] Make sure emails display the right name/gender/information about family
- [X] Add a new box in the Profile page of a contact to display all the relationships the contact has
- [x] Remove all the unused methods about Offsprings/Progenitors
- [x] Fix unit tests
- [x] Change the API
- [x] Change the seeders
- [ ] Impact on account export.
- [ ] Impact on importing data with `vCard` and `.csv` files.
- [ ] Impact on account reset and deletion.
- [ ] [CHANGELOG](https://github.com/monicahq/monica/blob/master/CHANGELOG) entry added, if necessary, under `UNRELEASED`.
- [ ] If it's relevant, add the documentation about your feature in the README file.
- [ ] Add screenshots

### CODE REVIEW

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:8O3lzuzMr3MGsDm]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:j5pIrfOdousmq6K]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:rPoCH8wnqwUnaWl]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:hDdgh6HsflHZQ1S]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:vFzIjMNl6Oj8wa8]

**Comment by asbiin** [comment]
This PR would close #117 too right ?

**Comment by djaiss** [comment]
@asbiin yes it would close https://github.com/monicahq/monica/issues/117. Good catch!

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:ieXz5YmKWZJaE2C]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:gH81xJccyKUDwum]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:m8UXM8Q91nBTOxK]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:OILunfqrZnyM9tV]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:FsDSfjopII9mEuW]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:FXoXW6S0m2RSfxs]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:o1v8ayPMlppBXjv]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:QZeWGbXBhofNTEX]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:z1dFFPcTpMckFEB]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:dbcNLlSUq3sA8bb]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:QK8q9M58MD59qAe]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:WPnSpqpYuwcO3zt]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:F5JN4pGcaVXDBkh]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:YK1CMJl190t84Ng]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:pObSFdg5FHEszoc]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:dbXE1IHGmOg9FSs]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-test_it_displays_reminder_rules-0.png
[id:DxbgfLRHY9LIh9s]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-testBarCodeContent-0.png
[id:b8U9904JuPkppiV]

File: app/Account.php [hunk 1]
```diff
@@ -580,6 +603,76 @@ public function replaceGender(Gender $genderToDelete, Gender $genderToReplaceWit
...
+     * @param string $email
+     * @param string $password
+     * @return this
+     */
+    public static function createDefault($first_name, $last_name, $email, $password)
+    {
+        // create new account
+        $account = new self;
+        $account->api_key = str_random(30);
+        $account->created_at = Carbon::now();
```
Comment (asbiin) [inline app/Account.php]: you should use `now()` helper - see #1051

File: app/Account.php [hunk 2]
```diff
@@ -580,6 +603,76 @@ public function replaceGender(Gender $genderToDelete, Gender $genderToReplaceWit
...
+
+    /**
+     * Gets the RelationshipType object matching the given type.
+     *
+     * @param  string $relationshipTypeName
+     * @return RelationshipType
+     */
+    public function getRelationshipTypeByType(string $relationshipTypeName)
+    {
+        return $this->relationshipTypes->where('name', $relationshipTypeName)->first();
```
Comment (asbiin) [inline app/Account.php]: add `->get()` to return the object, and not the `Query\Builder` instance

File: resources/views/people/relationship/index.blade.php [hunk 3]
```diff
@@ -1,70 +1,43 @@
...
-          <a href="#" onclick="if (confirm('{{ trans('people.significant_other_unlink_confirmation') }}')) { $(this).closest('.sidebar-box-paragraph').find('.entry-delete-form').submit(); } return false;" class="action-link {{ $contact->id }}-unlink-relationship">
-            {{ trans('app.remove') }}
-          </a>
+   @include('people.relationship._relationship', ['relationships' => $familyRelationships])
 
-          <form method="POST" action="{{ route('people.relationships.unlink', [$contact, $partner]) }}" class="entry-delete-form hidden">
-            {{ csrf_field() }}
-          </form>
+   <p class="mb0">
+    <a href="/people/{{ $contact->id }}/relationships/new?type={{ $contact->account->getRelationshipTypeByType('child')->id }}">Add</a>
```
Comment (asbiin) [inline resources/views/people/relationship/index.blade.php]: "Add" should be localized

File: resources/views/people/relationship/index.blade.php [hunk 4]
```diff
@@ -1,70 +1,43 @@
...
-    <p class="sidebar-box-paragraph">
-      <a href="{{ route('people.relationships.add', $contact) }}">{{ trans('people.significant_other_cta') }}</a>
-    </p>
+   @include('people.relationship._relationship', ['relationships' => $friendRelationships])
 
-  @endif
+   @include('people.relationship._relationship', ['relationships' => $workRelationships])
 
+   <p class="mb0">
+    <a href="/people/{{ $contact->id }}/relationships/new?type={{ $contact->account->getRelationshipTypeByType('friend')->id }}">Add</a>
```
Comment (asbiin) [inline resources/views/people/relationship/index.blade.php]: "Add" should be localized

File: app/Account.php (hunk 2 above)
Comment (djaiss) [reply app/Account.php]: `get` would return a collection, I need a single item.

File: resources/views/people/relationship/index.blade.php (hunk 4 above)
Comment (djaiss) [reply resources/views/people/relationship/index.blade.php]: Well done

File: resources/views/people/relationship/index.blade.php (hunk 3 above)
Comment (djaiss) [reply resources/views/people/relationship/index.blade.php]: Well done

File: app/Account.php (hunk 1 above)
Comment (djaiss) [reply app/Account.php]: done

File: app/Account.php (hunk 2 above)
Comment (asbiin) [reply app/Account.php]: ok ok

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-testBarCodeContent-0.png
[id:5fAGKBJKfcyN2SZ]

**Pull request merged**